salesforce_bulk/tests/test_salesforce_bulk.py

```diff
@@ -288,6 +288,61 @@ def test_query(self):
             ['Email', 'Id', 'Name']
         )
 
+    def test_query_pk_chunk(self):
+        bulk = SalesforceBulk(self.sessionId, self.endpoint)
+        self.bulk = bulk
+
+        job_id = bulk.create_query_job("Contact", contentType=self.contentType, pk_chunking=True)
+        self.jobs.append(job_id)
+        self.assertIsNotNone(re.match("\w+", job_id))
+
+        query = "Select Id,Name,Email from Contact"
+        batch_id = bulk.query(job_id, query)
+        self.assertIsNotNone(re.match("\w+", batch_id))
+
+        try:
+            i = 0
+            while not bulk.is_batch_done(batch_id):
+                print("Job not done yet...")
+                print(bulk.batch_status(batch_id))
+                time.sleep(2)
+                i += 1
+                if i == 20:
+                    raise Exception
+        except BulkBatchFailed as e:
+            if e.state != bulk_states.NOT_PROCESSED:
+                raise
+
+        batches = bulk.get_batch_list(job_id)
+        print (batches)
+        batch_ids = [x['id'] for x in batches if x['state'] != bulk_states.NOT_PROCESSED]
+        requests = [bulk.get_query_batch_request(x, job_id) for x in batch_ids]
+        print (requests)
+        for request in requests:
+            self.assertTrue(request.startswith(query))
+
+        all_results = []
+
+        i = 0
+        while not all(bulk.is_batch_done(i, job_id) for i in batch_ids):
```

Comment:
Unintentional reuse of the `i` variable?

Reply:
ugh, that's what i get for using a single letter variable :(